### DISCUSSION TITLE
Computing inertia and hydrostatic stiffness matrices?

### DISCUSSION THREAD

**Discussion by maha-haji** [capytaine/capytaine#55] [General]
Hi,

Can one compute the inertia and hydrostatic stiffness matrices using capytaine? For instance, the example at https://ancell.in/capytaine/latest/user_manual/cookbook.html#animation-of-the-rao indicates that: The computation of the RAO requires the values of the inertia matrix and the hydrostatic stiffness matrix. They need to be computed independently.

Thanks!

**Comment by mancellin** [comment]
Hi,

No, the computation of these matrices is not implemented in Capytaine. I would like to add them in the future, but for now you will need to figure out another way.

In the example you linked, I've used [meshmagick](https://github.com/LHEEA/meshmagick).
If other users have other suggestions, here is a good place to share them.

**Comment by RubendeBruin** [comment]
Hi @maha-haji , you may want to have a look at DAVE. https://open-ocean.org.
(Disclaimer: I'm the author)

Scene.dynamics_M and Scene.dynamics_K will give you the inertia and stiffness matrices.

**Comment by maha-haji** [comment]
Hi @mancellin,

Thanks for the tip regarding [meshmagick](https://github.com/LHEEA/meshmagick). Is there a straightforward way to use it with FloatingBodies generated by Capytaine?

Thanks!

**Comment by mancellin** [comment]
Yes, you can pass the vertices and faces arrays directly from one to the other. Try something like that:
```python
import meshmagick
mm_mesh = meshmagick.Mesh(body.mesh.vertices, body.mesh.faces, name=body.mesh.name)
```
where `body` is a `FloatingBody` from Capytaine.
